ext/url/00_url.js

```diff
@@ -319,16 +319,24 @@
      * @param {string} base
      */
     constructor(url, base = undefined) {
-      const prefix = "Failed to construct 'URL'";
-      url = webidl.converters.DOMString(url, { prefix, context: "Argument 1" });
-      if (base !== undefined) {
-        base = webidl.converters.DOMString(base, {
+      if (url?.[_url] && base === undefined) {
+        this[webidl.brand] = webidl.brand;
+        this[_url] = { ...url[_url] };
```

Comment:
This is observable if someone either modified `URL.prototype.toString()`, or added a `toString()` method to the URL instance.

Reply:
unsure i understand what you mean

Reply:
```ts
URL.prototype.toString = function () { console.log("hello"); return this.href };
console.log("1");
const url = new URL("https://deno.land");
console.log("2");
const url2 = new URL(url);
console.log("3");
```

This prints `1\n2\nhello\n3\n` right now. With your patch this would print `1\n2\n3\n` - that is a problem.

There are two ways to tackle this problem:

1. You condition your fast path on `url.toString === URLPrototypeToString` (it is unmodified).
2. Instead of this change you add a `href -> components` global LRU cache. This would actually be really really really useful for HTTP server perf where you'll have many `new URL(req.url)` with the same `req.url` string.

Reply:
@lucacasonato I went with number 2. PTAL